src/main/java/com/ncov/module/service/MaterialRequiredService.java

```diff
@@ -82,6 +84,38 @@
         return carry(materialRequiredEntities);
     }
 
+    @Transactional(rollbackFor = Exception.class)
+    public MaterialResponse update(Long materialId
+            , MaterialRequest material
+            , Long userId) {
+        MaterialRequiredEntity presentMaterial = getById(materialId);
+        if (!Objects.equals(presentMaterial.getMaterialRequiredUserId(), userId)) {
+            throw new AccessDeniedException("permission denied!");
+        }
+        MaterialDto materialDto = material.getMaterials().get(0);
+        AddressDto address = material.getAddress();
+        MaterialRequiredEntity entity = MaterialRequiredEntity.builder()
+                .id(materialId)
+                .country(address.getCountry())
+                .province(address.getProvince())
+                .city(address.getCity())
+                .district(address.getDistrict())
+                .streetAddress(address.getStreetAddress())
+                .materialRequiredContactorName(material.getContactorName())
+                .materialRequiredContactorPhone(material.getContactorPhone())
+                .materialRequiredOrganizationName(material.getOrganisationName())
+                .materialRequiredComment(material.getComment())
+                .materialRequiredImageUrls(ImageUtils.joinImageUrls(materialDto.getImageUrls()))
+                .materialRequiredName(materialDto.getName())
+                .materialRequiredCategory(materialDto.getCategory())
+                .materialRequiredQuantity(materialDto.getQuantity())
+                .materialRequiredStandard(materialDto.getStandard())
+                .gmtModified(new Date())
```

Comment:
This will overwrite `gmtCreated` as null which should not be modified once persisted.